io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -38,6 +40,7 @@ module Control.Monad.IOSim.Internal (
   Trace (..),
   TraceEvent (..),
   liftST,
+  liftStStm,
```

Comment:
I understand that the `-Stm` suffix is consistent with the naming used in the `StmA` data type, but looking at this export list, `liftStStm` seems rather inconsistent: `ST` and `STM` are always uppercased in the other identifiers. Nevertheless, I admit that `liftSTSTM` is no good either and `liftStSTM` seems internally inconsistent :slightly_smiling_face:.

So because I can't think of a better name, I don't object.

However, as I comment below, why do you need it? If we can avoid it, we could also avoid the naming problem :smile:

Reply:
I removed `liftStStm` in 30 Dec force push.